File: app/src/main/java/cn/ifafu/ifafu/data/http/parser/LoginParamParser.java
package cn.ifafu.ifafu.data.http.parser;

import java.util.Map;

import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import okhttp3.ResponseBody;

public class LoginParamParser extends ParamsParser {
    @Override
    public ObservableSource<Map<String, String>> apply(Observable<ResponseBody> upstream) {
        return upstream.map(responseBody -> parse(responseBody.string()));
    }
}

File: app/src/main/java/cn/ifafu/ifafu/data/http/parser/LoginParamParser.kt
package cn.ifafu.ifafu.data.http.parser

import io.reactivex.Observable
import io.reactivex.ObservableSource
import okhttp3.ResponseBody

class LoginParamParser : ParamsParser() {

    override fun apply(upstream: Observable<ResponseBody>): ObservableSource<MutableMap<String, String>> {
        return upstream.map { responseBody -> parse(responseBody.string()) }
    }
}
